### DISCUSSION TITLE
Api Guard not working for me.

### DISCUSSION THREAD

**Discussion by pk4all** [adonisjs/core#1273] [Help] [answered]
I am using adonis 5 for my APP Api
I am using API guard and for token I am using database driver.
My app setting is in follow screens... 
Please help me what is wrong with me...
Thanks.
![2020-07-20 (2)](https://user-images.githubusercontent.com/12910992/87914946-19c9a180-ca8f-11ea-946d-3eac3e546f88.png)
![2020-07-20 (3)](https://user-images.githubusercontent.com/12910992/87914949-1a623800-ca8f-11ea-974c-2fa352081114.png)
![2020-07-20 (4)](https://user-images.githubusercontent.com/12910992/87914953-1aface80-ca8f-11ea-934d-d73f7088d8eb.png)
![2020-07-20 (6)](https://user-images.githubusercontent.com/12910992/87914958-1c2bfb80-ca8f-11ea-92dc-49cdc6fe3434.png)
![2020-07-20](https://user-images.githubusercontent.com/12910992/87914961-1cc49200-ca8f-11ea-80ec-8b31850a7f41.png)
![2020-07-20 (1)](https://user-images.githubusercontent.com/12910992/87914939-17ffde00-ca8f-11ea-9f69-12f0d81a651d.png)

**Comment by thetutlage** [comment]
Please share the code examples inline in this thread in order to help you further.

**Reply by pk4all** [reply]
My guard and auth config sode is ...

interface GuardsList {
    web: {
      implementation: SessionGuardContract<'user', 'web'>,
      config: SessionGuardConfig<'user'>,
    },
    api: {
      implementation: OATGuardContract<'user', 'api'>,
      config: OATGuardConfig<'user'>,
    },
  }

Config code is ..

api: {
      driver: 'oat',
      tokenProvider: {
        driver: 'database',
        table: 'api_tokens',
      },

      provider: {
        driver: 'lucid',
        identifierKey: 'id',
        uids: ['email'],
        model: User,
      },
    },



My login code is...


try {
      const data=request.all()
      const email = data.email
      const password = data.password
      const token = await auth.use('api').attempt(email, password)
      return token.toJSON()
     // return response.status(200).send(auth.user)
    } catch (error) {
      return response.status(201).send(error.code)
    }

I am getting bad field error.

**Reply by thetutlage** [reply]
Hey, here's how you can format your codeblocks. https://guides.github.com/features/mastering-markdown/#GitHub-flavored-markdown.

Please note, if you will create threads with proper information, then it will increase the chances of getting answered. Otherwise, we will continue with the questions/answers dance.

**Reply by pk4all** [reply]
Ok thanks for your guideline. I have found my error. I have missed a column name (expires_at) in my DB. Actually in new framework doc I have not founded token table schema.

And sorry for my bad code writing.

**Comment by McSneaky** [comment, marked as answer]
UsersController.ts line 82 seems to be the place of return currently?
Return / log / debug out whole error stack in there, not just error code. Error stack or message might have some useful information

**Reply by pk4all** [reply]
Ok thanks for your guideline. I have found my error. I have missed a column name (expires_at) in my DB. Actually in new framework doc I have not founded token table schema.
